symengine/dict.h

```diff
@@ -9,6 +9,8 @@
 #define SYMENGINE_DICT_H
 
 #include <gmpxx.h>
+#include <piranha/mp_integer.hpp>
+#include <piranha/hash_set.hpp>
```

Comment:
Yes, the point is not to use Piranha at all in the header file. It requires a redesign a bit.